python/cudf/cudf/core/column/string.py

```diff
@@ -3896,6 +3896,86 @@ def startswith(self, pat: Union[str, Sequence]) -> SeriesOrIndex:
 
         return self._return_or_inplace(result_col)
 
+    def removesuffix(self, suffix: str) -> SeriesOrIndex:
+        """
+        Remove a suffix from an object series.
+
+        If the suffix is not present, the original string will be returned.
+
+        Parameters
+        ----------
+        suffix : str
+            Remove the suffix of the string.
+
+        Returns
+        -------
+        Series/Index: object
+            The Series or Index with given suffix removed.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> s = cudf.Series(["foo_str", "bar_str", "no_suffix"])
+        >>> s
+        0    foo_str
+        1    bar_str
+        2    no_suffix
+        dtype: object
+        >>> s.str.removesuffix("_str")
+        0    foo
+        1    bar
+        2    no_suffix
+        dtype: object
+        """
+        if suffix is None or len(suffix) == 0:
+            return self._return_or_inplace(self._column)
+        ends = self.endswith(suffix)
+        removed = self.slice(0, -len(suffix))
+        result = cudf._lib.copying.copy_if_else(
+            column.as_column(removed), self._column, column.as_column(ends)
+        )
+        return self._return_or_inplace(result)
```

Comment:
```suggestion
        if suffix is None or len(suffix) == 0:
            return self._return_or_inplace(self._column)

        ends_column = libstrings.endswith(
                self._column, cudf.Scalar(suffix, "str")
         )
        removed_column = libstrings.slice_strings(self._column, start=0, stop=-len(suffix), step=None)
        result = cudf._lib.copying.copy_if_else(
            removed_column, self._column, ends_column
        )
        return self._return_or_inplace(result)
```

We could directly call the libcudf functions and work with those `columns`, this way we can avoid assigning `index` objects from the results of `self.endswith` or `self.slice`(as `self._return_or_inplace` will assign parents index).

Reply:
style check will probably fail for my suggestion. Needs some formatting.